### PR TITLE
feat(alias): Add aliases for cache delete (del) and clear (reset)

### LINKED ISSUE #109: add an "clear()" method (an alias of "reset()") to make the API compatible with Map
The current API is inconsistent with the [`Map.prototype.clear()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/clear). This results in an error when using `lru-cache` in places where `Map`-like object is expected, e.g. in the case of using `lru-cache` with [DataLoader](https://github.com/facebook/dataloader).

### PR DESCRIPTION
**Pull request by prashanthr**
Addresses / Fixes #109 
### Summary
- Added aliases `delete` (`del`) and `clear` (`reset`)
- Included tests
- Updated Documentation

### CODE REVIEW

**Comment by prashanthr** [comment]
FYI The `expires` test in the forEach tap test module is very unreliable. On different build attempts, it produced different results. I you notice my last 2 commits, I made no change to the test module and it produced different test run results. I'd suggest refactoring this in the future.

**Comment by prashanthr** [comment]
PR ready to go

**Comment by sergey-nagaytsev** [comment]
delete() does not comply with https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/delete - it does not return any boolean value.

**Comment by prashanthr** [comment]
@sergey-nagaytsev I've just created an alias of the existing functionality
`lrucache.del` does not return a value and I've simply mapped `lrucache.delete` to `lrucache.del`. If the maintainer / reviewer wants to make that change to comply with the global Map behavior, I'd welcome it.

File: test/alias.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+})
+
+test('alias clear', function (t) {
+  var cache1 = new LRU({max: 10})
+  var cache2 = new LRU({max: 10})
+  cache1.set('key', 'value')
+  cache2.set('key', 'value')
+  cache1.reset()
+  cache2.clear()
+  t.equal(cache1.get('key'), undefined)
```
Comment (GrayedFox) [inline test/alias.js]: I'm likely being overzealous here but shouldn't we expect the length of `cache1` and `cache2` to be zero after clearing/resetting (and not just that they old key is now gone?)

Comment (prashanthr) [reply test/alias.js]: @GrayedFox You are correct. That probably should be an additional test.

**Pull request closed without merging**